ntbs-service/Services/SearchService.cs

```diff
@@ -1,62 +1,51 @@
-            INtbsSearchBuilder secondQueryable, PaginationParameters paginationParameters);
+        Task<(IList<int> notificationIds, int count)> OrderAndPaginateQueryableAsync(INtbsSearchBuilder firstQueryable,
+            PaginationParameters paginationParameters, ClaimsPrincipal user);
     }
 
     public class SearchService : ISearchService
     {
+        private readonly IUserService _userService;
 
-        public IQueryable<Notification> FilterById(IQueryable<Notification> notifications, string IdFilter) 
+        public SearchService(IUserService userService)
         {
-            int.TryParse(IdFilter, out int parsedIdFilter);
-            return notifications.Where(s => s.NotificationId.Equals(parsedIdFilter) 
-                    || s.ETSID.Equals(IdFilter) || s.LTBRID.Equals(IdFilter) || s.PatientDetails.NhsNumber.Equals(IdFilter));
+            _userService = userService;
         }
 
-        public IQueryable<Notification> FilterByPartialDate(IQueryable<Notification> notifications, PartialDate partialDate) 
-        {
-            partialDate.TryConvertToDateTimeRange(out DateTime? dateRangeStart, out DateTime? dateRangeEnd);
-            return notifications.Where(s => s.PatientDetails.Dob >= dateRangeStart && s.PatientDetails.Dob < dateRangeEnd);
-        }
-
-        public IQueryable<Notification> FilterBySex(IQueryable<Notification> notifications, int sexId) 
-        {
-            return notifications.Where(s => s.PatientDetails.SexId.Equals(sexId));
-        }
-
-        public async Task<(IList<int> notificationIds, int count)> OrderAndPaginateQueryablesAsync(
+        public async Task<(IList<int> notificationIds, int count)> OrderAndPaginateQueryableAsync(
             INtbsSearchBuilder firstBuilder,
-            INtbsSearchBuilder secondBuilder,
-            PaginationParameters paginationParameters)
+            PaginationParameters paginationParameters,
+            ClaimsPrincipal user)
         {
-            var notificationIdsQueryable = OrderQueryableByNotificationDate(firstBuilder.GetResult())
-                                                                .Union(OrderQueryableByNotificationDate(secondBuilder.GetResult()));
-
+            
+            var notificationIdsQueryable = await OrderQueryableByEditPermissionThenNotificationDateAsync(firstBuilder.GetResult(), user);
             var notificationIds = await GetPaginatedItemsAsync(notificationIdsQueryable.Select(n => n.NotificationId), paginationParameters);
             var count = await notificationIdsQueryable.CountAsync();
             return (notificationIds, count);
         }
 
-        private IQueryable<Notification> OrderQueryableByNotificationDate(IQueryable<Notification> query) 
+        private async Task<IQueryable<Notification>> OrderQueryableByEditPermissionThenNotificationDateAsync(IQueryable<Notification> query, ClaimsPrincipal user) 

```

Comment:
As discussed in person, an option here to prevent passing in every single tbService is for National team users to not run the `.Contains()` logic... but not likely to pose too much of a difference so very happy to be ignored.